modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/ServerStatisticsIntegrationTest.java

```diff
@@ -0,0 +1,640 @@
+ * limitations under the License.
+ *
+ */
+
+package org.apache.ignite.internal.processors.query.calcite.integration;
+
+import java.math.BigInteger;
+import java.sql.Date;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.cache.QueryEntity;
+import org.apache.ignite.cache.query.annotations.QuerySqlField;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.processors.query.QueryEngine;
+import org.apache.ignite.internal.processors.query.calcite.CalciteQueryProcessor;
+import org.apache.ignite.internal.processors.query.calcite.QueryChecker;
+import org.apache.ignite.internal.processors.query.calcite.util.Commons;
+import org.apache.ignite.internal.processors.query.h2.IgniteH2Indexing;
+import org.apache.ignite.internal.processors.query.stat.IgniteStatisticsManager;
+import org.apache.ignite.internal.processors.query.stat.StatisticsKey;
+import org.apache.ignite.internal.processors.query.stat.StatisticsTarget;
+import org.apache.ignite.internal.processors.query.stat.config.StatisticsObjectConfiguration;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.junit.Test;
+
+/**
+ * Tests for server side statistics usage.
+ */
+public class ServerStatisticsIntegrationTest extends AbstractBasicIntegrationTest {
+    /** Server instance. */
+    private IgniteEx srv;
+
+    /** All types table row count. */
+    private static final int ROW_COUNT = 100;
+
+    /** All types table nullable fields. */
+    private static final String[] NULLABLE_FIELDS = {
+        "boolean_obj_field",
+        "short_obj_field",
+        "integer_field",
+        "long_obj_field",
+        "float_obj_field",
+        "double_obj_field",
+    };
+
+    /** All types table non nullable fields. */
+    String[] NON_NULLABLE_FIELDS = {
+        "short_field",
+        "int_field",
+        "long_field",
+        "float_field",
+        "double_field"
+    };
+
+    /** All types table numeric fields. */
+    private static final String[] NUMERIC_FIELDS = {
+        "short_obj_field",
+        "integer_field",
+        "long_obj_field",
+        "float_obj_field",
+        "double_obj_field",
+        "short_field",
+        "int_field",
+        "long_field",
+        "float_field",
+        "double_field"
+    };
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        createAndPopulateAllTypesTable(0, ROW_COUNT);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected int nodeCount() {
+        return 1;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() {
+        cleanQueryPlanCache();
```

Comment:
why do you clean cache here? I comment this part and still all ok here.

Reply:
Just to simplify debug in case when some queries will be the same in different tests. To avoid thinking about cached values, but without cluster restarts. U suggest to remove it?